File: backend/app/schemas/crm.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.lead import LeadRead


class AiFilterResponse(BaseModel):
    with_website: list[LeadRead]
    without_website: list[LeadRead]


class CrmAddRequest(BaseModel):
    lead_ids: list[str]


class CrmAddResult(BaseModel):
    added: int
    skipped: int


class CrmEntryRead(BaseModel):
    id: str
    lead: LeadRead
    category: str
    added_at: datetime


class CrmEntryListResponse(BaseModel):
    items: list[CrmEntryRead]
    total: int

File: frontend/src/lib/api/crm.ts
import { apiFetch } from './client';
import type { LeadDTO } from './leads';

export interface AiFilterResponseDTO {
  with_website: LeadDTO[];
  without_website: LeadDTO[];
}

export function aiFilterLeads(token: string | null) {
  return apiFetch<AiFilterResponseDTO>('/api/v1/leads/ai-filter', token);
}

export interface CrmAddResultDTO {
  added: number;
  skipped: number;
}

export function addToCrm(token: string | null, leadIds: string[]) {
  return apiFetch<CrmAddResultDTO>('/api/v1/crm/entries', token, {
    method: 'POST',
    body: JSON.stringify({ lead_ids: leadIds })
  });
}

export interface CrmEntryDTO {
  id: string;
  lead: LeadDTO;
  category: 'with_website' | 'no_website';
  added_at: string;
}

export interface CrmEntryListResponseDTO {
  items: CrmEntryDTO[];
  total: number;
}

export function listCrmEntries(token: string | null) {
  return apiFetch<CrmEntryListResponseDTO>('/api/v1/crm/entries', token);
}
